watcher/src/verification_reports_collector.rs

```diff
@@ -117,6 +109,75 @@ impl NodeClient for ConsensusNodeClient {
     }
 }
 
+fn verification_report_from_node_url(
```

Comment:
Most of this logic is taken from https://github.com/mobilecoinfoundation/mobilecoin/blob/7b68eaae98e390bcca638769079b89cec0c476c2/connection/src/thick.rs#L302

Unfortunately I didn't see a test for the report behavior of the `ConsensusNodeClient` or one for the `ThickClient::attest()`.

The test in this module tests the `VerificationReportsCollector` with a fake `NodeClient`.

To test this out I ran it against the testnet nodes as documented, https://github.com/mobilecoinfoundation/mobilecoin/blob/7b68eaae98e390bcca638769079b89cec0c476c2/watcher/README.md?plain=1#L4.